lib/SPIRV/libSPIRV/SPIRVInstruction.h

```diff
@@ -577,9 +577,15 @@ class SPIRVStore : public SPIRVInstruction, public SPIRVMemoryAccess {
     SPIRVInstruction::validate();
     if (getSrc()->isForward() || getDst()->isForward())
       return;
-    assert(getValueType(PtrId)->getPointerElementType() ==
-               getValueType(ValId) &&
-           "Inconsistent operand types");
+    assert(
+        (getValueType(PtrId)
+             ->getPointerElementType()
+             ->isTypeUntypedPointerKHR() ||
+         // TODO: This check should be removed once we support untyped
+         // variables.
+         getValueType(ValId)->isTypeUntypedPointerKHR() ||
```

Comment:
Currently there is a bug, this check allows to store a pointer with one storage class into a pointer (to a pointer) with the base pointer type of a different storage class, which is invalid.